google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -93,12 +97,302 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket pb) {
+
+      public Boolean getIsLive() {
+        return isLive;
+      }
+
+      public List<StorageClass> getMatchesStorageClass() {
+        return matchesStorageClass;
+      }
+
+      /** Builder for {@code LifecycleCondition}. */
+      public static class Builder {
+        private Integer age;
+        private DateTime createdBefore;
+        private Integer numberOfNewerVersions;
+        private Boolean isLive;
+        private List<StorageClass> matchesStorageClass;
+
+        private Builder() {}
+
+        /** Sets the age in days a Blob should be for an Action to be executed **/
+        public Builder setAge(Integer age) {
+          this.age = age;
+          return this;
+        }
+
+        /** Sets the date a Blob should be created before for an Action to be executed. Note that only the date
+         * will be considered, if the time is specified it will be truncated **/
+        public Builder setCreatedBefore(DateTime createdBefore) {
+          this.createdBefore = createdBefore;
+          return this;
+        }
+
+        /** Sets the number of newer versions a Blob should have for an Action to be executed **/
+        public Builder setNumberOfNewerVersions(Integer numberOfNewerVersions) {
+          this.numberOfNewerVersions = numberOfNewerVersions;
+          return this;
+        }
+
+        /** Sets whether a Live Blob meets the Condition for an Action to be executed. If this is set to {@code false},
+         * the condition is met by Archived Blobs.
+         */
+        public Builder setIsLive(Boolean live) {
+          this.isLive = live;
+          return this;
+        }
+
+        /** Sets a list of Storage Classes ({@code StorageClass} objects) for a Blob to match for an Action to be executed **/
+        public Builder setMatchesStorageClass(List<StorageClass> matchesStorageClass) {
+          this.matchesStorageClass = matchesStorageClass;
+          return this;
+        }
+
+        /** Creates a {@code LifecycleCondition} object **/
+        public LifecycleCondition build() {
+          return new LifecycleCondition(this);
+        }
+      }
+    }
+
+    /**
+     * Base class for the Action to take when a Lifecycle Condition is met. Specific Actions are expressed as subclasses
+     * of this class, accessed by static factory methods
+     */
+    public abstract static class LifecycleAction implements Serializable {
+      private static final long serialVersionUID = 5801228724709173284L;
+
+      public abstract String getActionType();
+
+      /** Creates a new {@code DeleteLifecycleAction}. Blobs that meet the Condition associated with this action will be
+       * deleted
+       */
+      public static DeleteLifecycleAction newDeleteAction() {
+        return new DeleteLifecycleAction();
+      }
+
+      /** Creates a new {@code SetStorageClassLifecycleAction}. Blobs that meet the Condition associated with this
+       * action will have their Storage Class changed to the one specified
```

Comment:
Change: `Blobs that meet the Condition associated with this action will have their Storage Class changed to the one specified`
To: `A Blob's storage class that meets the action's conditions will be changed to the specified storage class.`